test/extensions/filters/network/ssh/wire/field_test.cc

```diff
@@ -0,0 +1,735 @@
+  EXPECT_EQ(TypeParam{}, f.value);
+  f.value = random_value<TypeParam>();
+  EXPECT_EQ(f.value, f.operator*());
+  EXPECT_EQ(&f.value, f.operator->());
+  TypeParam t = f;
+  EXPECT_EQ(t, *f);
+  EXPECT_EQ(t, f.value);
+  EXPECT_EQ(t, f);
+  EXPECT_EQ(f, f);
+  EXPECT_EQ(std::strong_ordering::equal, t <=> f.value);
+  EXPECT_EQ(std::strong_ordering::equal, t <=> f);
+
+  t = random_value<TypeParam>();
+  f = t;
+  EXPECT_EQ(t, f.value);
+  EXPECT_EQ(t, f.operator*());
+  EXPECT_EQ(&f.value, f.operator->());
+
+  f = random_value<TypeParam>();
+
+  field<TypeParam> f2{f};
+  EXPECT_EQ(f2.value, f.value);
+  EXPECT_NE(&f2.value, &f.value);
+
+  field<TypeParam> f3{std::move(f2)};
+  EXPECT_EQ(f3.value, f.value);
+
+  field<TypeParam> f4;
+  f4 = f;
+  EXPECT_EQ(f4.value, f.value);
+  EXPECT_NE(&f4.value, &f.value);
+
+  field<TypeParam> f5;
+  f5 = std::move(f4);
+  EXPECT_EQ(f5.value, f.value);
+}
+
+TYPED_TEST(FieldTest, Operators_ListTypes) {
+  if constexpr (requires { typename TypeParam::value_type; }) {
+    field<TypeParam> f;
+    while (f.value.empty()) {
+      f = random_value<TypeParam>();
+    }
+    EXPECT_EQ(f.value[0], f[0]);
+    EXPECT_EQ(&f.value[0], &f[0]);
+  }
+}
+
+TYPED_TEST(FieldTest, Decode) {
+  Buffer::OwnedImpl buf;
+  auto value = random_value<TypeParam>();
+  write(buf, value);
+
+  field<TypeParam> t;
+  auto len = buf.length();
+  auto r = t.decode(buf, len);
+  EXPECT_TRUE(r.ok());
+  EXPECT_EQ(len, *r);
+  EXPECT_EQ(value, t.value);
+}
+
+TYPED_TEST(FieldTest, Decode_Error) {
+  Buffer::OwnedImpl buf;
+  field<TypeParam> t;
+  auto r = t.decode(buf, 4);
+  EXPECT_FALSE(r.ok());
+  EXPECT_EQ(absl::StatusCode::kInvalidArgument, r.status().code());
+  EXPECT_EQ("short read", r.status().message());
+}
+
+TYPED_TEST(FieldTest, Encode) {
+  Buffer::OwnedImpl buf;
+  auto value = random_value<TypeParam>();
+  write(buf, value);
+  auto encoded = flushTo<bytes>(buf);
+
+  Buffer::OwnedImpl buf2;
+  field<TypeParam> f;
+  f = value;
+  auto r = f.encode(buf2);
+  EXPECT_TRUE(r.ok());
+  EXPECT_EQ(encoded.size(), *r);
+  EXPECT_EQ(encoded, flushTo<bytes>(buf2));
+}
+
+template <typename T>
+class DecodeMsgTest : public testing::Test {
+public:
+  void SetUp() override {
+    expected_type = random_value<SshMessageType>();
+    expected_value = random_value<T>();
+    write(buffer, expected_type);
+    write(buffer, expected_value);
+  }
+
+  Buffer::OwnedImpl buffer;
+  SshMessageType expected_type;
+  T expected_value;
+};
+
+template <typename T>
+class EncodeMsgTest : public testing::Test {
+public:
+  void SetUp() override {
+    expected_type = random_value<SshMessageType>();
+    field = random_value<T>();
+  }
+
+  Buffer::OwnedImpl buffer;
+  SshMessageType expected_type;
+  field<T> field;
+};
+
+TYPED_TEST_SUITE(DecodeMsgTest, BasicFieldTypes);
+TYPED_TEST_SUITE(EncodeMsgTest, BasicFieldTypes);
+
+TYPED_TEST(DecodeMsgTest, Decode) {
+  field<TypeParam> field;
+  auto len = this->buffer.length();
```

Comment:
I also get some size_t / uint64_t errors for calls to `decodeMsg()`.

```suggestion
  size_t len = this->buffer.length();
```

(same on lines 148, 156, and 166)